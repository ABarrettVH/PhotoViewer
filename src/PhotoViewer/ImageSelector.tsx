import React from "react";
import './ImageSelector.css'



export function ImageSelector(props: {srcUrl: string, setNewUrl: (url: string) => void}) {
    const imageUrls = getImageUrls();
     return (
        <div className="imageSelector">
            {imageUrls.map((image) => (
                <button onClick = {()=> props.setNewUrl(image)}>
                    <img 
                        className="galleryItem"
                        src={image}
                    />
                </button>
            ))}
        </div>


     );
}
const brokenImages = [
    1, 24, 32, 36, 44, 47
];




function getImageUrls() {
    const urls = [];

    for (let i = 0; i < 50; i++) {
        if (!brokenImages.includes(i)) {
            const imageNumberString = i.toString().padStart(2, '0');
            urls.push(`https://picsum.photos/id/6${imageNumberString}/400/300.jpg`)
        }
    }

    return urls;
}

export const imageUrls = getImageUrls();