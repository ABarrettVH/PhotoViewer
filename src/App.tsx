import { useState } from 'react'
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'
import { PhotoViewer } from './PhotoViewer/PhotoViewer'
import './PhotoViewer/PhotoViewer.css'
import { ImageSelector, imageUrls } from './PhotoViewer/ImageSelector'

function App() {

  return (
      <div>
          <h1>React Photo Viewer</h1>
          <PhotoViewer src={"https://picsum.photos/id/600/1600/900.jpg"}/>
          <ImageSelector/>
      </div>
  )
}

export default App
