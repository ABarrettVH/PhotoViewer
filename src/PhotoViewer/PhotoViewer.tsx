import React from "react";
import './PhotoViewer.css'
import { useState } from "react";


export function PhotoViewer(props) {
    const [url, newUrl ] = useState(props.src);



    return (
        <div>
            <img  className="topImage" src={url} ></img>
        </div>
    );
}