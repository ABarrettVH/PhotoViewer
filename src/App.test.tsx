import React, { useReducer } from 'react';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import App from './App';
import { PhotoViewer } from './PhotoViewer/PhotoViewer';
import { getImageUrls, ImageSelector } from './PhotoViewer/ImageSelector';
import renderer from 'react-test-renderer';

test('renders the title- React Photo Viewer', () => {
  const { getByText } = render(<App />);
  const textElement = getByText(/React Photo Viewer/i);
  expect(textElement).toBeVisible();
});

test('Renders an image given a url', () => {
  const { getByAltText } = render(<PhotoViewer srcUrl='https://picsum.photos/id/600/1600/900.jpg' />);
  
  const imgElement = getByAltText(/dynamic top image/i);
  expect(imgElement).toBeVisible();
});

test('Check urls array doesnt contain broken images', () => {
  const imageUrls = getImageUrls();
  
  const brokenImage = !imageUrls.includes('https://picsum.photos/id/601/400/300.jpg');

  expect(brokenImage).toBeTruthy();
});


test('Check first picture is as expected', () => {
  const imageUrls = getImageUrls();
  
  const firstImage = imageUrls[0];

  expect(firstImage).toBe('https://picsum.photos/id/600/400/300.jpg');
});


describe("App", () => {
 it("Should match snapshot", () => {

 const tree = render(<App/>);

 expect(tree).toMatchSnapshot();
 });
});



describe("My Component", () => {

  it('renders correctly', async () => {
  const {getAllByAltText} = render(<App/>);

  const topImages = getAllByAltText('dynamic top image'); 
  const topImage = topImages[0] as HTMLImageElement;
  const galleryPictures = getAllByAltText('button');
  const buttonToClick = galleryPictures[4] as HTMLImageElement;
  
  expect(topImage.src).toBe("https://picsum.photos/id/600/1600/900.jpg")

  await fireEvent.click(buttonToClick);

  expect(topImage.src).toBe("https://picsum.photos/id/605/400/300.jpg");

})
})



