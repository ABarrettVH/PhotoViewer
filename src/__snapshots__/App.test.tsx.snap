// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1>
          React Photo Viewer
        </h1>
        <div>
          <img
            alt="dynamic top image"
            class="topImage"
            src="https://picsum.photos/id/600/1600/900.jpg"
          />
        </div>
        <div
          class="imageSelector"
        >
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/600/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/602/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/603/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/604/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/605/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/606/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/607/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/608/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/609/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/610/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/611/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/612/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/613/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/614/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/615/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/616/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/617/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/618/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/619/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/620/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/621/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/622/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/623/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/625/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/626/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/627/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/628/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/629/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/630/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/631/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/633/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/634/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/635/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/637/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/638/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/639/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/640/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/641/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/642/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/643/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/645/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/646/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/648/400/300.jpg"
            />
          </button>
          <button>
            <img
              alt="button"
              class="galleryItem"
              src="https://picsum.photos/id/649/400/300.jpg"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        React Photo Viewer
      </h1>
      <div>
        <img
          alt="dynamic top image"
          class="topImage"
          src="https://picsum.photos/id/600/1600/900.jpg"
        />
      </div>
      <div
        class="imageSelector"
      >
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/600/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/602/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/603/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/604/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/605/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/606/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/607/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/608/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/609/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/610/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/611/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/612/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/613/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/614/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/615/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/616/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/617/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/618/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/619/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/620/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/621/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/622/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/623/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/625/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/626/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/627/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/628/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/629/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/630/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/631/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/633/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/634/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/635/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/637/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/638/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/639/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/640/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/641/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/642/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/643/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/645/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/646/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/648/400/300.jpg"
          />
        </button>
        <button>
          <img
            alt="button"
            class="galleryItem"
            src="https://picsum.photos/id/649/400/300.jpg"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
